codec-http/src/main/java/io/netty/handler/codec/http/HttpContentDecoder.java

```diff
@@ -214,20 +214,21 @@ public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
     private void cleanup() {
         if (decoder != null) {
             // Clean-up the previous decoder if not cleaned up correctly.
-            if (decoder.finish()) {
-                for (;;) {
-                    ByteBuf buf = decoder.readInbound();
-                    if (buf == null) {
-                        break;
-                    }
-                    // Release the buffer
-                    buf.release();
-                }
-            }
+            decoder.finishAndReleaseAll();
             decoder = null;
         }
     }
 
+    private void cleanupSafely(ChannelHandlerContext ctx) {
+        try {
+            cleanup();
+        } catch (Exception e) {
+            // If cleanup throws any error we need to propagate it through the pipeline and then ensure we call
+            // super.channelInactive(...) so the event is not lost.
```

Comment:
this is also used for `handlerRemoved` ... should this comment be more generic (e.g. "so we don't fail to propagate pipeline events...")

Reply:
Yep let me adjust it.